File: shellSort.js
function shellSort(arr) {
  var increment = arr.length / 2;
  var pasada = 1;
  console.log("Arreglo Desordenado: "+arr)
  while (increment > 0) {
      for (i = increment; i < arr.length; i++) {
          var j = i;
          var temp = arr[i];
  
          while (j >= increment && arr[j-increment] > temp) {
              arr[j] = arr[j-increment];
              j = j - increment;
            }
            
            arr[j] = temp;
          }
          
          console.log("Pasada Numero "+ pasada + ": " +arr)
          pasada++
          if (increment == 2) {
            increment = 1;
      } else {
          increment = parseInt(increment*5 / 11);
      }
  }
return arr;
}

console.log("Arreglo Ordenado: "+shellSort([5,2,4,6]))

File: shellsort.py
def shellSort(arr): 
  
    n = len(arr) 
    gap = n//2
  
   
    while gap > 0: 
  
        for i in range(gap,n): 
  
            temp = arr[i] 
  
            j = i 
            while  j >= gap and arr[j-gap] >temp: 
                arr[j] = arr[j-gap] 
                j -= gap 
            arr[j] = temp 
        gap = gap//2

arr = [ 12, 34, 54, 2, 3,11,10,9,8,6,98,78,984,94,949,4,944,9494] 
  
n = len(arr) 
print ("Array before sorting:") 
for i in range(n): 
    print(arr[i]), 
  
shellSort(arr) 
  
print ("\nArray after sorting:") 
for i in range(n): 
    print(arr[i]), 
  

